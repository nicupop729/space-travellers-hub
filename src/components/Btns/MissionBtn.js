import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMissionAction } from '../../redux/missions/missions';

const MissionBtn = ({ className, text, id }) => {
  const dispatch = useDispatch();
  return (
    <button
      type="button"
      onClick={() => dispatch(joinMissionAction(id))}
      className={className}
    >
      {text}
    </button>
  );
};

MissionBtn.propTypes = {
  className: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default MissionBtn;
