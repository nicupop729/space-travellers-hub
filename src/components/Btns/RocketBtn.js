import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { reserveRocketAction } from '../../redux/rockets/rockets';

const RocketBtn = ({ className, text, id }) => {
  const dispatch = useDispatch();
  return (
    <button
      type="button"
      onClick={() => dispatch(reserveRocketAction(id))}
      className={className}
    >
      {text}
    </button>
  );
};

RocketBtn.propTypes = {
  className: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
};

export default RocketBtn;
