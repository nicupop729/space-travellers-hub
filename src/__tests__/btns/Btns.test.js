import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import MissionBtn from '../../components/Btns/MissionBtn';
import RocketBtn from '../../components/Btns/RocketBtn';

describe('snapshot testing', () => {
  test('snapshot for MissionBtn component', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <MissionBtn />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });

  test('snapshot for RocketBtn component', () => {
    const rendererComponent = renderer
      .create(
        <Provider store={store}>
          <RocketBtn />
        </Provider>,
      )
      .toJSON();
    expect(rendererComponent).toMatchSnapshot();
  });
});
