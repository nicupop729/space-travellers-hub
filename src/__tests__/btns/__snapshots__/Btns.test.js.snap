// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for MissionBtn component 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;

exports[`snapshot testing snapshot for RocketBtn component 1`] = `
<button
  onClick={[Function]}
  type="button"
/>
`;
